src/lib/y2storage/proposal_settings.rb

```diff
@@ -24,58 +22,113 @@
 require "y2storage/subvol_specification"
 require "y2storage/filesystems/type"
-
-Yast.import "ProductFeatures"
+require "y2storage/partitioning_features"
 
 module Y2Storage
+  # Class to manage settings used by the proposal (typically read from control.xml)
   #
-  # User-configurable settings for the storage proposal.
-  # Those are settings the user can change in the UI.
-  #
-  class ProposalUserSettings
-    include Yast::Logger
+  # When a new object is created, all settings are nil or [] in case of a list is
+  # expected. See {#for_current_product} to initialize settings with some values.
+  class ProposalSettings
     include SecretAttributes
+    include PartitioningFeatures
 
-    VALID_DELETE_MODES = [:none, :all, :ondemand]
-    private_constant :VALID_DELETE_MODES
-
+    # @note :legacy format
     # @return [Boolean] whether to use LVM
     attr_accessor :use_lvm
 
+    # @note :legacy format
     # @return [Filesystems::Type] type to use for the root filesystem
     attr_accessor :root_filesystem_type
-    #
+
+    # @note :legacy format
     # @return [Boolean] whether to enable snapshots (only if Btrfs is used)
     attr_accessor :use_snapshots
 
+    # @note :legacy format
     # @return [Boolean] whether to propose separate partition/volume for /home
     attr_accessor :use_separate_home
 
+    # @note :legacy format
     # @return [Filesystems::Type] type to use for the home filesystem, if a
     #   separate one is proposed
     attr_accessor :home_filesystem_type
 
+    # @note :legacy format
     # @return [Boolean] whether to enlarge swap based on the RAM size, to ensure
     #   the classic suspend-to-ram works
     attr_accessor :enlarge_swap_for_suspend
 
+    # @note :legacy format
     # @return [String] device name of the disk in which / must be placed. If set
     #   to nil, the proposal will try to find a good candidate
     attr_accessor :root_device
 
+    # @note :legacy format
+    # @return [Boolean] whether to resize Windows systems if needed
+    attr_accessor :resize_windows
+
+    # @note :legacy format
+    # @return [DiskSize] root size used when calculating the :min size for
+    #   the proposal.
+    attr_accessor :root_base_size
+
+    # @note :legacy format
+    # @return [DiskSize] root size used when calculating the :desired size for
+    #   the proposal.
+    attr_accessor :root_max_size
+
+    # @note :legacy format
+    # @return [Numeric] used to adjust size when distributing extra space
+    attr_accessor :root_space_percent
+
+    # @note :legacy format
+    # @return [Numeric] used to adjust size when using snapshots
+    attr_accessor :btrfs_increase_percentage
+
+    # @note :legacy format
+    # @return [DiskSize] min disk size to allow a separate home
+    attr_accessor :min_size_to_use_separate_home
+
+    # @note :legacy format
+    # @return [String] default btrfs subvolume path
+    attr_accessor :btrfs_default_subvolume
+
+    # @note :legacy format
```

Comment:
In fact, this is not a proposal setting. This setting is used by the installer at the end of installation, but it doesn't affect the proposal.

Reply:
So it's not used at all in the current `Y2Storage` and we should take the opportunity to remove it.